pkg/mountutil/volumestore/volumestore.go

```diff
@@ -83,13 +83,29 @@ func (vs *volumeStore) Create(name string, labels []string) (*native.Volume, err
 	}
 	volPath := filepath.Join(vs.dir, name)
 	volDataPath := filepath.Join(volPath, DataDirName)
-	fn := func() error {
+	volFilePath := filepath.Join(volPath, volumeJSONFileName)
+	fn := func() (err error) {
 		if err := os.Mkdir(volPath, 0700); err != nil {
 			return err
 		}
+		defer func() {
+			if err != nil {
+				// ignore cleanup error
+				os.Remove(volPath)
+			}
+		}()
 		if err := os.Mkdir(volDataPath, 0755); err != nil {
 			return err
 		}
+		defer func() {
+			if err != nil {
+				// ignore cleanup error
+				if _, serr := os.Stat(volFilePath); serr == nil || !os.IsNotExist(serr) {
```

Comment:
If stat returns error for the file, it should print error and shouldn't continue cleanup.

Reply:
> it should print error and shouldn't continue cleanup.

`volFilePath` does not exist and `volDataPath` exists  => we should continue cleanup

just drop `stat` and log remove error should be fine

Reply:
> If stat returns error for the file, it should print error and shouldn't continue cleanup.

Fixed.